apigentools/commands/generate.py

```diff
@@ -31,6 +31,157 @@
+    ),
+    help="Path to directory with downstream templates (default: '{}')".format(
+        constants.DEFAULT_DOWNSTREAM_TEMPLATES_DIR
+    ),
+)
+@click.option(
+    "--git-email",
+    help="Email of the user to author git commits as. Note this will permanently"
+    " modify the local repos git config to use this author",
+    default=env_or_val("APIGENTOOLS_GIT_AUTHOR_EMAIL", None),
+)
+@click.option(
+    "--git-name",
+    help="Name of the user to author git commits as. Note this will permanently"
+    " modify the local repos git config to use this author",
+    default=env_or_val("APIGENTOOLS_GIT_AUTHOR_NAME", None),
+)
+@click.option(
+    "-t",
+    "--template-dir",
+    default=env_or_val("APIGENTOOLS_TEMPLATES_DIR", constants.DEFAULT_TEMPLATES_DIR),
+    help="Path to directory with processed upstream templates (default: '{}')".format(
+        constants.DEFAULT_TEMPLATES_DIR
+    ),
+)
+@click.option(
+    "--builtin-templates",
+    is_flag=True,
+    default=False,
+    help="Use unpatched upstream templates",
+)
+@click.option(
+    "-T",
+    "--templates-output-dir",
+    default=env_or_val("APIGENTOOLS_TEMPLATES_DIR", constants.DEFAULT_TEMPLATES_DIR),
+    help="Path to directory where to put processed upstream templates (default: {})".format(
+        constants.DEFAULT_TEMPLATES_DIR
+    ),
+)
+@click.option(
+    "--templates-source",
+    type=click.Choice(
+        [
+            constants.TEMPLATES_SOURCE_LOCAL_DIR,
+            constants.TEMPLATES_SOURCE_OPENAPI_GIT,
+            constants.TEMPLATES_SOURCE_OPENAPI_JAR,
+            constants.TEMPLATES_SOURCE_SKIP,
+        ],
+        case_sensitive=False,
+    ),
+    default=env_or_val("APIGENTOOLS_TEMPLATES_SOURCE", constants.TEMPLATES_SOURCE_SKIP),
+    help="Source to use for obtaining templates to be processed (default: 'skip')",
+)
+@click.option(
+    "-p",
+    "--template-patches-dir",
+    default=env_or_val(
+        "APIGENTOOLS_TEMPLATE_PATCHES_DIR", constants.DEFAULT_TEMPLATE_PATCHES_DIR
+    ),
+    help="Directory with patches for upstream templates (default: '{}')".format(
+        constants.DEFAULT_TEMPLATE_PATCHES_DIR
+    ),
+)
+@click.option(
+    "--jar-path",
+    nargs=1,
+    default=env_or_val("APIGENTOOLS_OPENAPI_JAR", constants.OPENAPI_JAR),
+    help="Path to openapi-generator jar file (use if --templates-source=openapi-jar)",
+)
+@click.option(
+    "-u",
+    "--repo_url",
+    default=constants.OPENAPI_GENERATOR_GIT,
+    help="URL of the openapi-generator repo (default: '{}'; use if --templates-source=openapi-git)".format(
+        constants.OPENAPI_GENERATOR_GIT
+    ),
+)
+@click.option(
+    "--git-committish",
+    default="master",
+    nargs=1,
+    help="Git 'committish' to check out before obtaining templates "
+    "(default: 'master'; use if --templates-source=openapi-git)",
+)
+@click.pass_obj
+def generate(ctx, **kwargs):
+    """Generate client code"""
+    x = click.get_current_context()
```

Comment:
why?

Reply:
Leftover from a test I was running. Good catch, will remove it. (I need the context a bit later in the function to be able to invoke the template command. I was playing around with the context up here during testing and forgot to remove it)